### PR TITLE
Override toString(), hashCode() and equals() methods in ObserveCompositeRequest

### PR DESCRIPTION
**Pull request by JaroslawLegierski**
Added toString, hashCode and equals to ObserveCompositeRequest

### CODE REVIEW

**Comment by adamsero** [comment]
Fixed toString(), should be good to merge

**Comment by sbernard31** [comment]
On so small PR/changes, do  not hesitate squash all you commit in one.

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/ObserveCompositeRequest.java [hunk 1]
```diff
@@ -136,4 +136,46 @@ public ContentFormat getResponseContentFormat() {
     public Map<String, String> getContext() {
         return context;
     }
+
+    @Override
+    public final String toString() {
+        return String.format(
+                "ObserveCompositeRequest [paths=%s requestContentFormat=%s responseContentFormat=%s, context=%s]",
```
Comment (sbernard31) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/ObserveCompositeRequest.java:143]: I see there is a `,` just after `responseContentFormat=%s` but not after the others.

So I checked in code base and there is case where `,` is used and so other where it is not used.
But when you generate `toString` with eclipse, `,` is used.

So I will add it

**Comment by sbernard31** [comment]
Integrated in master (commit ecc113c4351fc351b9d300030e24b9c801be2737)

**Pull request closed without merging**